translator/translate/otel/receiver/otlp/translator.go

```diff
@@ -56,18 +62,20 @@ func WithDataType(dataType component.DataType) Option {
 var _ common.Translator[component.Config] = (*translator)(nil)
 
 func NewTranslator(opts ...Option) common.Translator[component.Config] {
-	t := &translator{factory: otlpreceiver.NewFactory()}
-	for _, opt := range opts {
-		opt.apply(t)
-	}
-	return t
+	return NewTranslatorWithName("", opts...)
 }
 
 func NewTranslatorWithName(name string, opts ...Option) common.Translator[component.Config] {
-	t := &translator{name: name, factory: otlpreceiver.NewFactory()}
+	t := &translator{name: name, index: -1, factory: otlpreceiver.NewFactory()}
```

Comment:
what does the index do here?

Reply:
The index allows us to distinguish between different otlp receivers. In this case, if you specify a fixed name to your receiver there is no need to track it with a index.

Reply:
oh so it's really just used for otlp/1, otlp/2..etc..? Should we just have it be instanceName or something? seems misleading

Reply:
You can have multiple otlp running on the same instance. I'd rather have a hash or something instead of InstanceName

Reply:
yeah it just seems confusing. I'd rather have it called `instanceNum` or something to refer it to the instance name.